### PR TITLE
#160 let's get rid of List.add() - Part 3

### PR DESCRIPTION
**Pull request by super132**
For task #160, introducing Transform for eliminating the use of `List.add()` in `RsWithHeaders`. This PR is split from #183

### CODE REVIEW

**Comment by super132** [comment]
@yegor256 , A iterable transform class `Transform` is introduced to replace the use of `List.add()` in `RsWithHeaders` with the conjunction use of `Concat`.

**Comment by super132** [comment]
@davvd , This PR is split from #183 , not sure if it needs to be reviewed by architect again. Please advise. Thanks.

**Comment by davvd** [comment]
@super132 Thanks for your pull request, let me find someone who can review it

**Comment by davvd** [comment]
@darkled please review,thanks

File: src/main/java/org/takes/rs/RsWithHeaders.java [hunk 1]
```diff
@@ -52,6 +52,10 @@ public RsWithHeaders(final Response res, final CharSequence... headers) {
         this(res, Arrays.asList(headers));
     }
 
+    //@todo #160:DEV To implement the concatenation and
```
Comment (darkled) [inline src/main/java/org/takes/rs/RsWithHeaders.java]: @super132 let's move this into the javadoc below

**Comment by darkled** [comment]
@super132 a minor comment above and there are some failed tests

**Comment by super132** [comment]
@darkled , I moved the codes into the javadoc. Please have a check. Thanks.
The test cases failure was not from my code and the codes in master branch contains a PMD error.

**Comment by darkled** [comment]
@super132 could you please fix this PMD? otherwise it would be impossible to merge your changes.

**Comment by super132** [comment]
@darkled, the build is fixed by merging the latest commits from master. Please have a check. Thanks.

**Comment by darkled** [comment]
@super132 thanks

**Comment by darkled** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by darkled** [comment]
@davvd we're waiting for #253 here

**Comment by davvd** [comment]
> @davvd we're waiting for #253 here

@darkled right, let's wait for #253

**Comment by davvd** [comment]
> @davvd we're waiting for #253 here

@darkled of course, let's wait for #253

**Comment by davvd** [comment]
> @davvd we're waiting for #253 here

@darkled yep, got it, let's wait for #253 being solved

**Comment by davvd** [comment]
> @davvd we're waiting for #253 here

@darkled got it, let's wait for #253

**Comment by davvd** [comment]
> @davvd we're waiting for #253 here

@darkled yes, waiting for #253

**Comment by super132** [comment]
@darkled , I saw pull requests can be merged now. Shall we try merging again? Thanks.

**Comment by darkled** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

File: src/main/java/org/takes/rs/RsWithHeaders.java [hunk 2]
```diff
@@ -54,6 +54,12 @@ public RsWithHeaders(final Response res, final CharSequence... headers) {
 
     /**
      * Ctor.
+     * @todo #160:DEV To implement the concatenation and
```
Comment (darkled) [inline src/main/java/org/takes/rs/RsWithHeaders.java]: @super132 please add an estimation here.
>ERROR: "puzzle src/main/java/org/takes/rs/RsWithHeaders.java:55-60 has an estimate of 0 minutes, which is lower than 15 minutes"

**Comment by super132** [comment]
@darkled , I have added the estimate here. Please have a check. Thanks.

**Comment by darkled** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@darkled
According to our [quality rules](http://www.teamed.io/qa.html):
 > Comments were mostly about design problems, not cosmetic issues.
 
 There were mostly cosmetic issues found during review. Try to pay attention to bigger design problems and the overall quality of the code.
Please confirm that this new rule is clear.

**Comment by darkled** [comment]
@elenavolokhova I confirm, thank you

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks a lot, next time everybody should try to make it better

**Comment by davvd** [comment]
@darkled 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 57164971... **23 mins** sent to your balance (ID `57164982`), many thanks! It took 167 hours and 31 mins.... review comments (c=8) added as a bonus... +23 added to your rating, at the moment it is: [+2062](http://www.netbout.com/b/35091?open=rating)

**Pull request merged**